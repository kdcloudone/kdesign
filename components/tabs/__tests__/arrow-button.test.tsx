import React from 'react'
import { render, mount } from 'enzyme'
import mountTest from '../../../tests/shared/mountTest'
import ArrowButton from '../arrow-button'

describe('Tabs Arrow Button', () => {
  mountTest(ArrowButton)

  it('renders correctly', () => {
    const leftWrapper = render(<ArrowButton direction="left" />)
    const rightWrapper = render(<ArrowButton direction="right" />)
    expect(leftWrapper).toMatchSnapshot()
    expect(rightWrapper).toMatchSnapshot()
  })

  it('should clickable', () => {
    const onClickLeft = jest.fn()
    const onClickRight = jest.fn()
    const leftWrapper = mount(<ArrowButton direction="left" onClick={onClickLeft} />)
    const rightWrapper = mount(<ArrowButton direction="right" onClick={onClickRight} />)
    leftWrapper.find('.kd-arrow-button-left').simulate('click')
    rightWrapper.find('.kd-arrow-button-right').simulate('click')
    expect(onClickLeft).toHaveBeenCalled()
    expect(onClickRight).toHaveBeenCalled()
  })

  it('should have displayName static property', () => {
    const wrapper = mount(<ArrowButton direction="left" />)
    expect((wrapper.type() as any).displayName).toBe('ArrowButton')
  })
})
