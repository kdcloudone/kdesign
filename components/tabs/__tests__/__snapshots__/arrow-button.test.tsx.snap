// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs Arrow Button renders correctly 1`] = `
<span
  class="kd-arrow-button kd-arrow-button-left"
>
  <i
    class="kdicon kdicon-arrow-left"
  />
</span>
`;

exports[`Tabs Arrow Button renders correctly 2`] = `
<span
  class="kd-arrow-button kd-arrow-button-right"
>
  <i
    class="kdicon kdicon-arrow-right"
  />
</span>
`;
